sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1799,7 +1807,9 @@ class SessionCatalog(
       case f if FunctionRegistry.functionSet.contains(f) => (f, "SYSTEM")
       case f if TableFunctionRegistry.functionSet.contains(f) => (f, "SYSTEM")
       case f => (f, "USER")
-    }.distinct
+    }.distinct.map { f =>
+      (f._1.copy(catalog = sessionCatalogOption(f._1.database)), f._2)
```

Comment:
I think we should only attach catalog name for user functions, not system (builtin) functions.

Reply:
it does not affect the result since the builtin function has no database, but I will change it to apply user functions for more clearer.